CHANGELOG.md

```diff
@@ -1,5 +1,126 @@
 # go-ipfs changelog
 
+### 0.4.3-rc1 - 2016-07-23
+
+This is a maintenance release which comes with a couple of nice enhancements, and improves the performance of Storage, Bitswap, as well as Content and Peer Routing. It also introduces a handful of new commands and options, and fixes a good bunch of bugs.
+
+This is the first Release Candidate. Unless there are vulnerabilities or regressions discovered, the final 0.4.3 release will happen about one week from now.
+
+- Security Vulnerability
+
+  - We are thankful that a group of independent researchers made us aware of a vulnerability which the `master` branch suffered from for about 3 weeks. It allowed an attacker to e.g. use an iframe to request malicious HTML and JS from the API of a local go-ipfs node, and use this to then extract the node's private key. We fixed this issue by reintroducing restrictions on which particular objects can be loaded through the API (@lgierth, [ipfs/go-ipfs#2949](https://github.com/ipfs/go-ipfs/pull/2949)), and by completely excluding the private key from the API (@Kubuxu, [ipfs/go-ipfs#2957](https://github.com/ipfs/go-ipfs/pull/2957)). We will also work on more hardening of the API in the next release.
```

Comment:
- Note that this was only in master and not in a release, 
- reiterate that if people stay on the releases as recommended they shouldn't have seen it at all.
- ask for more watchful eyes on security on PRs and merges